index.html

```diff
@@ -0,0 +1,464 @@
+              <dfn data-dfn-for="route to">auto</dfn>
+            </dt>
+            <dd>
+              The user agent's default launch routing behaviour is used.
+              <p class="note">
+                The user agent is expected to decide what works best for the
+                platform. E.g., on mobile devices that only support single app instances the user agent may
+                use `existing-client-navigate`, while on desktop devices that support
+                multiple windows the user agent may use `new-client` to avoid data loss.
+              </p>
+            </dd>
+            <dt>
+              <dfn data-dfn-for="route to">new-client</dfn>
+            </dt>
+            <dd>
+              A new web app client is created to load the launch's target URL.
+            </dd>
+            <dt>
+              <dfn data-dfn-for="route to">existing-client-navigate</dfn>
+            </dt>
+            <dd>
+              If an existing web app client is open it is brought to focus and
+              navigated to the launch's target URL. If there are no existing web
+              app clients the [=route to/new-client=] behaviour is used instead.
+            </dd>
+            <dt>
+              <dfn data-dfn-for="route to">existing-client-retain</dfn>
+            </dt>
+            <dd>
+              If an existing web app client is open it is brought to focus but
+              not navigated to the launch's target URL, instead the target URL
+              is communicated via {{LaunchParams}} . If there are no existing
+              web app clients the [=route to/new-client=] behaviour is used
+              instead.
+            </dd>
+          </dl>
+        </p>
+        <p>
+          To <dfn>process the `route_to` member</dfn>, given [=ordered
+          map=] |json_launch_handler:ordered map|, [=ordered map=]
+          |manifest_launch_handler:ordered map|, run the following:
+        </p>
+        <ol class="algorithm">
+          <li>If |json_launch_handler|["route_to"] does not [=map/exist=],
+          return.
+          </li>
+          <li>If the type of |json_launch_handler|["route_to"] is
+          [=list=]:
+            <ol class="algorithm">
+              <li>[=list/For each=] |entry| of |json_launch_handler|["route_to"]:
+                <ol class="algorithm">
+                  <li>If the type of |entry| is not [=string=], continue.
+                  </li>
+                  <li>If |json_launch_handler|["route_to"] is supported by the user agent, set manifest_launch_handler["route_to"] to |entry| and return.
+                  </li>
+                </ol>
+              </li>
+            </ol>
+          </li>
+          <li>If the type of |json_launch_handler|["route_to"] is not [=string=], return.
+          </li>
+          <li>If |json_launch_handler|["route_to"] is supported by the user agent, set manifest_launch_handler["route_to"] to |json_launch_handler|["route_to"].
```

Comment:
"..., and return."

Then, add an extra step:

"Set set manifest_launch_handler["route_to"] to "auto"."

So it is always guaranteed to have something set.

Oh, we need to deal with the early return on step 1 and 3. Either a) set mlh.rt to "auto" first, then have the later steps override it (messy), or b) instead of having step 1 and 3 as an early-exit, make them positive if statements which contain the rest of the algorithm inside them (my preference, since it's more functional programming style).

Reply:
Done.

Reply:
Still not quite right. The case where route_to is missing doesn't set the manifest copy to "auto", which it needs to do (so later code can assume it's set).

Either add that as an explicit action in step 1, or refactor so it doesn't early-return.